src/OrchardCore.Modules/OrchardCore.ContentFields/Services/DefaultContentPickerResultProvider.cs

```diff
@@ -36,11 +42,7 @@ public async Task<IEnumerable<ContentPickerResult>> Search(ContentPickerSearchCo
 
             foreach (var contentItem in contentItems)
             {
-                results.Add(new ContentPickerResult
-                {
-                    ContentItemId = contentItem.ContentItemId,
-                    DisplayText = contentItem.DisplayText
```

Comment:
I think the only necessary change is to replace this by 
```csharp
DisplayText = String.IsNullOrWhitespace(contentItem.DisplayText)
 ? T["{0}" - "1", contentItem.Id, contentItem.ContentType]
 : contentItem.DisplayText
 ;
```

Reply:
I should certainly be making use of the string localizer, I didn't think about that. But that suggested change alone would only affect the strings that are displayed as dropdown options. If you selected one of those options (such as one that has an empty `contentItem.DisplayText`), then save/publish and reload the editor, the selected string that should be displayed in the text field (before the dropdown options) will have a different format to the dropdown option display texts. In this case it will mean that no text is displayed in the text field at all (because `contentItem.DisplayText` is empty for the selected item). So on the face of it, it will look like nothing has been selected. Basically there are other places where the `ContentPickerResult` is constructed. I think the `DefaultContentPickerResultProvider` should expose a builder to make sure that the `ContentPickerResults` are displayed consistently.

This is the reason that I defined a generic version of the `IContentPickerResultProvider` interface, defining an additional `ContentPickerResult BuildResult(T item)` method. If the `DefaultContentPickerResultProvider` doesn't implement this, then there is no way for `ContentPickerFieldDisplayDriver` to know how to display the selected item (see commit https://github.com/OrchardCMS/OrchardCore/pull/2572/commits/9d18d6694a3b3134118bcd20964826a3e0c8fdb9)

------------------------------------------------------------------------

On another note, given that the text displayed in the dropdown would now display the content type (and not just the display text), I think that the content items should now be searchable by content type (and not just display text). This is the purpose of my last commit https://github.com/OrchardCMS/OrchardCore/pull/2572/commits/8521a37b752903f35820cc1ab12174874d97fcdb . If this wasn't the case, there would be no way for content items with empty `contentItem.DisplayText` properties to show up in the content item search results